BetterActivities plugin

D3SOX opened the pull request:
Closes https://github.com/Vencord/plugin-requests/issues/41
Closes https://github.com/Vencord/plugin-requests/issues/258

# Preview

![image](https://github.com/Vendicated/Vencord/assets/24937357/bcebf9be-d337-435e-be28-360e4b03675e)
![image](https://github.com/Vendicated/Vencord/assets/24937357/d2ba69cb-5e47-47da-9ffb-026aa5427c52)
![image](https://github.com/Vendicated/Vencord/assets/24937357/277f425f-65e7-4e25-ad98-ff61b4370f08)

# Settings

![image](https://github.com/Vendicated/Vencord/assets/24937357/a664a95b-51bb-437b-8e04-064267cc58b8)

D3SOX committed:
feat(memberListActivities): add new plugin

Vendicated commented on src/plugins/memberListActivities/index.tsx:
```diff
@@ -0,0 +1,123 @@
+
+                const addImage = (image: string, alt: string) => {
+                    if (image.startsWith("mp:external/")) {
+                        const externalLink = image.replace(/mp:external\/.{0,43}\//, "").replaceAll("https/", "https://");
```
this will expose your ip

D3SOX replied:
Hm right. I guess need to find out how Discord renders them

D3SOX replied:
Resolved.

D3SOX committed:
chore(memberListActivities): remove obsolete settings

D3SOX committed:
fix(memberListActivities): use discord proxy for external media

D3SOX committed:
fix(memberListActivities): support twitch and attachment images

D3SOX committed:
refactor(memberListActivities): simplify media link transformation

D3SOX committed:
feat(memberListActivities): support application icons

D3SOX committed:
feat(memberListActivities): improve patch

D3SOX committed:
fix(memberListActivities): make icons round

D3SOX committed:
docs(memberListActivities): update screenshot

D3SOX committed:
fix(memberListActivities): always replace icon

Update patch to support single app icons (like simple Minecraft RPC)

D3SOX committed:
fix(memberListActivities): use correct fetchApplication method

D3SOX committed:
feat(memberListActivities): configurable icon size

D3SOX committed:
Merge branch 'main' into plugin/memberListActivities

D3SOX committed:
feat(memberListActivities): option to disable rendering GIFs

D3SOX committed:
fix(memberListActivities): prefer large image

I found the large image to be more applicable most of the time as the small image is often only used as an indicator.

Maybe make this a setting?

I also thought about maintaining a list of applications where to use which image but gave up on that idea.

D3SOX committed:
feat(memberListActivities): better console icons support

D3SOX committed:
fix(memberListActivities): get rid of duplicate icons

D3SOX committed:
fix(memberListActivities): prefer application icons

D3SOX committed:
fix(memberListActivities): update patch

D3SOX committed:
Merge branch 'main' into plugin/memberListActivities

D3SOX committed:
feat(memberListActivities): show default icon when no icons where found

D3SOX committed:
Merge branch 'main' into plugin/memberListActivities

D3SOX committed:
feat(memberListActivities): add tooltips

D3SOX committed:
refactor(memberListActivities): don't spread activity into img element

D3SOX committed:
Merge branch 'refs/heads/main' into plugin/memberListActivities

D3SOX committed:
feat(memberListActivities): better tooltips

D3SOX committed:
chore(memberListActivities): linter be happy pls

D3SOX committed:
chore(memberListActivities): omg linter yes

like who cares

D3SOX committed:
feat(memberListActivities): show different asset in tooltip

D3SOX committed:
feat(memberListActivities): show time bar in tooltip

D3SOX committed:
fix(memberListActivities): add word-break to fix cut off text

D3SOX committed:
Merge branch 'refs/heads/main' into plugin/memberListActivities

D3SOX committed:
refactor(memberListActivities): move types to their own file

D3SOX committed:
chore(memberListActivities): fix typo in comment

D3SOX committed:
feat(memberListActivities): show elapsed time in tooltip and support twitch images

D3SOX committed:
feat(betterActivities): rename and add show all activities functionality

D3SOX commented:
This now closes two feature requests. Pls review :)

D3SOX committed:
refactor(betterActivities): add type for ActivityView component

D3SOX committed:
feat(betterActivities): allow showing all activities below each other

D3SOX committed:
fix(betterActivities): set default icon size to 15

idk why I've set this to 20

D3SOX committed:
feat(betterActivities): setting to not show special ones first

and reorganize/group the settings

D3SOX committed:
chore(betterActivities): add type for icon

D3SOX committed:
docs(betterActivities): don't include screenshots in git

D3SOX committed:
docs(betterActivities): add listview screenshot

Vendicated commented on src/plugins/betterActivities/index.tsx:
```diff
@@ -0,0 +1,524 @@
+/*
```
this is a very massive file. please split it up into a few separate files in a way that makes sense! don't overdo it, just like 3-4 files

D3SOX replied:
I've split it up into multiple files, is this enough?

Vendicated replied:
yep, already a lot better! just trying to avoid creating a second FakeNitro situation here
![image](https://github.com/Vendicated/Vencord/assets/45497981/2eb60fe9-14bd-4717-9e41-0abeafe2e808)

Vendicated commented on src/plugins/betterActivities/README.md:
```diff
@@ -0,0 +1,7 @@
+
+![Member List](https://github.com/Vendicated/Vencord/assets/24937357/4c034963-4448-483a-ba1d-c04f74e4aa51)
+![Multiple Activities](https://github.com/Vendicated/Vencord/assets/24937357/11a2f15e-59b0-4072-847a-33444d964674)
+![List view](https://github.com/Vendicated/Vencord/assets/24937357/277f425f-65e7-4e25-ad98-ff61b4370f08)
```
these alt texts are pretty meaningless. alt texts should serve as an alternative to the image, not just serve as a title for the image

```md
![Shows game icons next to the status line in the member list](...)
![Shows all activities in the profile via a carousel](...)
...
```

D3SOX replied:
Updated the alt texts

Vendicated reviewed (commented):
i'll do another review once the big file has been split into multiple smaller files

there are a lot of very long lines. try to keep your lines only so long that they still are fully visible with like 60% screen width on an average size monitor, so around 80-120 characters

D3SOX committed:
refactor(betterSettings): split index into multiple files

D3SOX committed:
docs(betterSettings): improve alt text

D3SOX committed:
refactor(betterSettings): re-use ActivityViewProps type

D3SOX committed:
Merge branch 'refs/heads/dev' into plugin/memberListActivities

D3SOX committed:
refactor(betterActivities): ActivityTooltip as function component

D3SOX committed:
refactor(betterActivities): remove obsolete restartNeeded props

D3SOX committed:
fix(betterActivities): brand-experiment is now brand-500

D3SOX committed:
fix(betterActivities): update icons patch

D3SOX committed:
Merge branch 'refs/heads/dev' into plugin/memberListActivities

D3SOX committed:
fix(betterActivities): show all activities for profiles v2

D3SOX committed:
fix(betterActivities): correct patch

D3SOX committed:
feat(betterActivities): setting for showing descriptions

D3SOX committed:
fix(betterActivities): use newer activity view

Discord removed the old ones now

D3SOX committed:
fix(betterActivities): correctly pass props

D3SOX committed:
fix(betterActivities): patch location including application prop

D3SOX committed:
fix(betterActivities): handle xbox edge case

D3SOX committed:
fix(betterActivities): handle fetchApplication errors

D3SOX committed:
fix(betterActivities): only patch BiteSizePopout type

We don't want to patch the user modal activity tab for example

D3SOX committed:
feat(betterActivities): use ActivityView in tooltip

D3SOX committed:
chore(betterActivities): remove unused code

D3SOX committed:
fix(betterActivities): avoid passing specific props to all activities

Now only the general props are used and the application is fetched for other activities

D3SOX committed:
fix(betterActivities): avoid conditional hook calls

Prevents crashing when the activities length changes

D3SOX committed:
chore: thx github web editor....

luximus-hunter commented:
What is currently needed to make this PR happen? If it's within my capabilities I'd love to help.

D3SOX committed:
Merge remote-tracking branch 'origin/main' into plugin/memberListActivities

D3SOX committed:
update to modern version pls look at this hello

D3SOX commented:
@Vendicated senpai notice me :3

D3SOX commented:
i should probably split this into multiple files again :o

D3SOX committed:
delete leftover file

D3SOX committed:
split into multiple files

D3SOX committed:
more refactoring

D3SOX committed:
fix(styles): correct class name

D3SOX committed:
chore(betterActivities): improve refactoring

D3SOX committed:
chore(betterActivities): revert stupid change
